packages/get-platform/src/getPlatform.ts

```diff
@@ -55,14 +55,27 @@ export async function getos(): Promise<GetOSResult> {
   }
 
   const distro = await resolveDistro()
+  const archFromUname = await getArchFromUname()
 
+  const usesOfficialEngines = !(
+    process.env.PRISMA_QUERY_ENGINE_BINARY ||
+    process.env.PRISMA_QUERY_ENGINE_LIBRARY ||
+    process.env.PRISMA_MIGRATION_ENGINE_BINARY
+  )
```

Comment:
Notes:
- `PRISMA_INTROSPECTION_ENGINE_BINARY` is now ignored
- We need to define the env vars as literals as we can't import `resolveBinary.ts` utils defined in `@prisma/internals` (it would add circular dependencies)